overlord/ifacestate/helpers.go

```diff
@@ -104,6 +104,17 @@ func (m *InterfaceManager) addSnaps() error {
 	if err != nil {
 		return err
 	}
+	// Before adding any snap scan the set of snaps we know about. If any of
+	// those is snapd then for the duration of this process always add implicit
+	// slots to snapd and not to any other type: os snap and use a mapper to
+	// use names core-snapd-system on state, in memory and in API responses,
+	// respectively.
+	for _, snapInfo := range snaps {
```

Comment:
I think doing this at the beginning of initialize would be more robust, otherwise we need to track exactly what is the first use of addImplicitSlots

Reply:
+1, I'll move it elsewhere.